### ISSUE TITLE
[Bug]: Apple M2  Chip, all other apps quit unexpectedly.

### ISSUE THREAD

**Issue by youthyan** [PlayCover/PlayCover#334]
### Describe the bug

Apple M2  Chip，
Except for Genshin Impact, which can be used normally. 
Other apps basically quit unexpectedly. 
Update playover 2.0, Except for Genshin Impact, all other apps quit unexpectedly.

### Steps to reproduce

<img width="920" alt="截屏2022-09-12 23 31 14" src="https://user-images.githubusercontent.com/105373147/189708560-1e49e577-1bbb-469b-9fff-d4d2fb9a4f7d.png">
<img width="354" alt="截屏2022-09-13 00 20 14" src="https://user-images.githubusercontent.com/105373147/189708603-e5b066e9-1ea3-4511-82f0-58f1aa71e63d.png">
<img width="873" alt="截屏2022-09-13 00 16 21" src="https://user-images.githubusercontent.com/105373147/189708694-81663a82-006a-40e4-b83f-f548c0370926.png">
<img width="973" alt="截屏2022-09-13 00 45 09" src="https://user-images.githubusercontent.com/105373147/189710675-49f4757a-eeeb-4aee-8a88-31a29a540129.png">

### Expected behaviour

Apple M2  Chip, Hopefully most apps will work without quitting unexpectedly

### Crash log

```shell
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               YouTube [27441]
Path:                  /Users/USER/Library/Containers/io.playcover.PlayCover/YouTube.app/YouTube
Identifier:            com.google.ios.youtube
Version:               17.33.2 (17.33.2)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2022-09-13 00:35:06.6600 +0800
OS Version:            macOS 12.5 (21G72)
Report Version:        12
Anonymous UUID:        6028AD84-C960-ACA4-F6B1-AEFE2A8FD9A1

Sleep/Wake UUID:       A97C4D41-5B35-4C7A-9D56-3E5682A0DB02

Time Awake Since Boot: 350000 seconds
Time Since Wake:       19714 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: '@rpath/widevine_cdm_secured_ios.framework/widevine_cdm_secured_ios'
Referenced from: '/Users/USER/Library/Containers/io.playcover.PlayCover/YouTube.app/YouTube'
Reason: tried: '' (mach-o file (/Users/liuyingyi/Library/Containers/io.playcover.PlayCover/YouTube.app/Frameworks/widevine_cdm_secured_ios.framework/widevine_cdm_secured_ios), but incompatible platform (have (iOS), need (MacCatalyst))), '' (mach-o file (/Users/liuyingyi/Library/Containers/io.playcover.PlayCover/YouTube.app/Frameworks/widevine_cdm_secured_ios.framework/widevine_cdm_secured_ios), but incompatible platform (have (iOS), need (MacCatalyst))), '' (no
(terminated at launch; ignore backtrace)

Thread 0 Crashed:
0   dyld                          	       0x103673e98 __abort_with_payload + 8
1   dyld                          	       0x10367b024 abort_with_payload_wrapper_internal + 104
2   dyld                          	       0x10367b058 abort_with_payload + 16
3   dyld                          	       0x103632a28 dyld4::halt(char const*) + 580
4   dyld                          	       0x10362dfb8 dyld4::prepare(dyld4::APIs&, dyld3::MachOAnalyzer const*) + 3644
5   dyld                          	       0x10362d06c start + 488


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000006   x1: 0x0000000000000001   x2: 0x000000016ddd69c8   x3: 0x00000000000000a6
    x4: 0x000000016ddd65c8   x5: 0x0000000000000000   x6: 0x0000000000000000   x7: 0x0000000000000ec0
    x8: 0x0000000000000020   x9: 0x0000000000000009  x10: 0x000000016ddd6635  x11: 0x00000000000000bb
   x12: 0x0000000000000000  x13: 0x0000000000000032  x14: 0x00000002739554f6  x15: 0x0000000000000000
   x16: 0x0000000000000209  x17: 0x000000010362b14c  x18: 0x0000000000000000  x19: 0x0000000000000000
   x20: 0x000000016ddd65c8  x21: 0x00000000000000a6  x22: 0x000000016ddd69c8  x23: 0x0000000000000001
   x24: 0x0000000000000006  x25: 0x000000016ddd65c8  x26: 0x0000000000000400  x27: 0x0000000000000290
   x28: 0x00000000000003a5   fp: 0x000000016ddd6590   lr: 0x000000010367b024
    sp: 0x000000016ddd6550   pc: 0x0000000103673e98 cpsr: 0x00001000
   far: 0x0000000103388000  esr: 0x56000080  Address size fault

Binary Images:
       0x103628000 -        0x103687fff dyld (*) <75627683-a780-32ad-ae34-cf86dd23a26b> /usr/lib/dyld

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=13.6M resident=0K(0%) swapped_out_or_unallocated=13.6M(100%)
Writable regions: Total=9232K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=9232K(100%)

                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
STACK GUARD                       56.0M        1 
Stack                             8176K        1 
VM_ALLOCATE                         16K        1 
__DATA                            5488K        3 
__DATA_CONST                        96K        1 
__LINKEDIT                        1856K        2 
__TEXT                            11.8M        2 
dyld private memory               1024K        1 
===========                     =======  ======= 
TOTAL                             84.1M       12 



-----------
Full Report
-----------

{"app_name":"YouTube","timestamp":"2022-09-13 00:35:06.00 +0800","app_version":"17.33.2","slice_uuid":"5ef0c340-3d84-37ff-9941-079fc815cd2d","build_version":"17.33.2","platform":6,"bundleID":"com.google.ios.youtube","share_with_app_devs":0,"is_first_party":0,"bug_type":"309","os_version":"macOS 12.5 (21G72)","incident_id":"ADEA96C9-5BE8-475D-86C9-C6B149C8800F","name":"YouTube"}
{
  "uptime" : 350000,
  "procLaunch" : "2022-09-13 00:35:06.5817 +0800",
  "procRole" : "Default",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "Mac14,7",
  "procStartAbsTime" : 8626975798644,
  "coalitionID" : 39990,
  "osVersion" : {
    "train" : "macOS 12.5",
    "build" : "21G72",
    "releaseType" : "User"
  },
  "captureTime" : "2022-09-13 00:35:06.6600 +0800",
  "incident" : "ADEA96C9-5BE8-475D-86C9-C6B149C8800F",
  "bug_type" : "309",
  "pid" : 27441,
  "procExitAbsTime" : 8626976332471,
  "translated" : false,
  "cpuType" : "ARM-64",
  "procName" : "YouTube",
  "procPath" : "\/Users\/USER\/Library\/Containers\/io.playcover.PlayCover\/YouTube.app\/YouTube",
  "bundleInfo" : {"CFBundleShortVersionString":"17.33.2","CFBundleVersion":"17.33.2","CFBundleIdentifier":"com.google.ios.youtube","DTAppStoreToolsBuild":"13F100"},
  "storeInfo" : {"deviceIdentifierForVendor":"C7B7A0AA-78A7-5721-AD78-D97A459011CE","thirdParty":true},
  "parentProc" : "launchd",
  "parentPid" : 1,
  "coalitionName" : "com.google.ios.youtube",
  "crashReporterKey" : "6028AD84-C960-ACA4-F6B1-AEFE2A8FD9A1",
  "wakeTime" : 19714,
  "sleepWakeUUID" : "A97C4D41-5B35-4C7A-9D56-3E5682A0DB02",
  "sip" : "enabled",
  "isCorpse" : 1,
  "exception" : {"codes":"0x0000000000000000, 0x0000000000000000","rawCodes":[0,0],"type":"EXC_CRASH","signal":"SIGABRT"},
  "termination" : {"code":1,"flags":518,"namespace":"DYLD","indicator":"Library missing","details":["(terminated at launch; ignore backtrace)"],"reasons":["Library not loaded: '@rpath\/widevine_cdm_secured_ios.framework\/widevine_cdm_secured_ios'","Referenced from: '\/Users\/USER\/Library\/Containers\/io.playcover.PlayCover\/YouTube.app\/YouTube'","Reason: tried: '' (mach-o file (\/Users\/liuyingyi\/Library\/Containers\/io.playcover.PlayCover\/YouTube.app\/Frameworks\/widevine_cdm_secured_ios.framework\/widevine_cdm_secured_ios), but incompatible platform (have (iOS), need (MacCatalyst))), '' (mach-o file (\/Users\/liuyingyi\/Library\/Containers\/io.playcover.PlayCover\/YouTube.app\/Frameworks\/widevine_cdm_secured_ios.framework\/widevine_cdm_secured_ios), but incompatible platform (have (iOS), need (MacCatalyst))), '' (no"]},
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":3169944,"threadState":{"x":[{"value":6},{"value":1},{"value":6138194376},{"value":166},{"value":6138193352},{"value":0},{"value":0},{"value":3776},{"value":32},{"value":9},{"value":6138193461},{"value":187},{"value":0},{"value":50},{"value":10529101046},{"value":0},{"value":521},{"value":4351766860,"symbolLocation":392,"symbol":"__simple_bprintf"},{"value":0},{"value":0},{"value":6138193352},{"value":166},{"value":6138194376},{"value":1},{"value":6},{"value":6138193352},{"value":1024},{"value":656},{"value":933}],"flavor":"ARM_THREAD_STATE64","lr":{"value":4352094244},"cpsr":{"value":4096},"fp":{"value":6138193296},"sp":{"value":6138193232},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":4352065176,"matchesCrashFrame":1},"far":{"value":4349001728}},"frames":[{"imageOffset":310936,"symbol":"__abort_with_payload","symbolLocation":8,"imageIndex":0},{"imageOffset":340004,"symbol":"abort_with_payload_wrapper_internal","symbolLocation":104,"imageIndex":0},{"imageOffset":340056,"symbol":"abort_with_payload","symbolLocation":16,"imageIndex":0},{"imageOffset":43560,"symbol":"dyld4::halt(char const*)","symbolLocation":580,"imageIndex":0},{"imageOffset":24504,"symbol":"dyld4::prepare(dyld4::APIs&, dyld3::MachOAnalyzer const*)","symbolLocation":3644,"imageIndex":0},{"imageOffset":20588,"symbol":"start","symbolLocation":488,"imageIndex":0}]}],
  "usedImages" : [
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 4351754240,
    "size" : 393216,
    "uuid" : "75627683-a780-32ad-ae34-cf86dd23a26b",
    "path" : "\/usr\/lib\/dyld",
    "name" : "dyld"
  }
],
  "sharedCache" : {
  "base" : 7443136512,
  "size" : 3144695808,
  "uuid" : "ffede77c-d90b-366b-96f2-665c2f50ee59"
},
  "vmSummary" : "ReadOnly portion of Libraries: Total=13.6M resident=0K(0%) swapped_out_or_unallocated=13.6M(100%)\nWritable regions: Total=9232K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=9232K(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nSTACK GUARD                       56.0M        1 \nStack                             8176K        1 \nVM_ALLOCATE                         16K        1 \n__DATA                            5488K        3 \n__DATA_CONST                        96K        1 \n__LINKEDIT                        1856K        2 \n__TEXT                            11.8M        2 \ndyld private memory               1024K        1 \n===========                     =======  ======= \nTOTAL                             84.1M       12 \n",
  "legacyInfo" : {
  "threadTriggered" : {

  }
},
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "607844aa04477260f58a8077",
      "factorPackIds" : {
        "SIRI_MORPHUN_ASSETS" : "6103050cbfe6dc472e1c982a"
      },
      "deploymentId" : 240000066
    },
    {
      "rolloutId" : "5fb4245a1bbfe8005e33a1e1",
      "factorPackIds" : {

      },
      "deploymentId" : 240000015
    }
  ],
  "experiments" : [

  ]
}
}

Model: Mac14,7, BootROM 7459.141.1, proc 8:4:4 processors, 8 GB, SMC 
Graphics: Apple M2, Apple M2, Built-In
Display: Color LCD, 2560 x 1600 Retina, Main, MirrorOff, Online
Memory Module: LPDDR5
AirPort: Wi-Fi, wl0: Mar 23 2022 20:47:07 version 18.60.27.0.7.8.129 FWID 01-bfc059c1
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB31Bus
USB Device: Extreme SSD
USB Device: USB31Bus
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.
```

### What version of PlayCover are you using?

1.1.1

### What version of macOS are you using?

Monterey (macOS 12)

### Issue Language

- [X] Yes my issue is written in English

**Comment by ZhichGaming** [comment]
Have you tried disabling sip and nvram?

**Comment by JoseMoreville** [comment]
Please, try using Stable build instead of the nightlies for this case

**Comment by youthyan** [comment]
>
I started with playover 1.1.1 and that was the case, then 2.0.
only Genshin Impact works.

>

**Comment by youthyan** [comment]
> 您是否尝试过禁用 sip 和 nvram？



> Only Genshin Impact works, of course try disabling sip and nvram

**Comment by youthyan** [comment]
<img width="256" alt="截屏2022-09-13 19 17 41" src="https://user-images.githubusercontent.com/105373147/189888622-8588ec7f-6261-4db9-a10e-67052a36f79d.png">
<img width="575" alt="截屏2022-09-13 19 17 03" src="https://user-images.githubusercontent.com/105373147/189888640-ca1ff485-a535-4ab6-b153-83da445c7bef.png">
Only Genshin Impact works, of course try disabling sip and nvram

**Comment by ohaiibuzzle** [comment]
Your YouTube IPA appears to be corrupted, as it is trying to load a dylib that doesn't exist (Check the `Frameworks` folder inside `YouTube.app`)

It is trying to load `widevine_cdm_secured_ios` and fail

**Comment by Wind-Explorer** [comment]
作为一款开源软件，自愿开发者们并无专为测试应用准备的设备。据我所知，我们开发团队中并无一人拥有M2系列的电脑，因此，很遗憾我们暂时无法对此问题进行深入研究，还请谅解。

译源：
https://cdn.discordapp.com/attachments/918276992812605541/1019208919333208064/unknown.png

**Comment by youthyan** [comment]
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: '@rpath/widevine_cdm_secured_ios.framework/widevine_cdm_secured_ios'
Referenced from: '/Users/USER/Library/Containers/io.playcover.PlayCover/YouTube.app/YouTube'
Reason: tried: '' (mach-o file (/Users/liuyingyi/Library/Containers/io.playcover.PlayCover/YouTube.app/Frameworks/widevine_cdm_secured_ios.framework/widevine_cdm_secured_ios), but incompatible platform (have (iOS), need (MacCatalyst))), '' (mach-o file (/Users/liuyingyi/Library/Containers/io.playcover.PlayCover/YouTube.app/Frameworks/widevine_cdm_secured_ios.framework/widevine_cdm_secured_ios), but incompatible platform (have (iOS), need (MacCatalyst))), '' (no
(terminated at launch; ignore backtrace)

**Comment by ZhichGaming** [comment]
Where did u get the ipa and how did u install?

**Comment by youthyan** [comment]
> 你从哪里得到的ipa，你是如何安装的？


https://ipa.playcover.workers.dev  
https://ipa.playcover.workers.dev/1:search?q=youtube 

我的app都是这里下载的。  双击就 默认使用playover 安装了

**Comment by youthyan** [comment]
> 你从哪里得到的ipa，你是如何安装的？

原神  我使用playover1.1.1 和2.0    都是一切正常。  
原神都是可以正常使用，一点问题都没有，
但是其他的apps，都不行

**Comment by ZhichGaming** [comment]
Can you try to reinstall the apps and maybe do a clean reinstall of PlayCover by deleting all the folders at these locations?

```

    "~/Library/Application Support/io.playcover.PlayCover",

    "~/Library/Caches/io.playcover.PlayCover",

    "~/Library/Containers/io.playcover.PlayCover",

    "~/Library/Frameworks/PlayTools.framework",

    "~/Library/Preferences/io.playcover.PlayCover.plist",

    "~/Library/Saved Application State/io.playcover.PlayCover.savedState",

    "~/Library/Application Support/me.playcover.PlayCover",

    "~/Library/Caches/me.playcover.PlayCover",

    "~/Library/Containers/me.playcover.PlayCover",

    "~/Library/Preferences/me.playcover.PlayCover.plist",

    "~/Library/Saved Application State/me.playcover.PlayCover.savedState",
```

**Comment by youthyan** [comment]
> Can you try to reinstall the apps and maybe do a clean reinstall of PlayCover by deleting all the folders at these locations?
> 
> ```
> 
>     "~/Library/Application Support/io.playcover.PlayCover",
> 
>     "~/Library/Caches/io.playcover.PlayCover",
> 
>     "~/Library/Containers/io.playcover.PlayCover",
> 
>     "~/Library/Frameworks/PlayTools.framework",
> 
>     "~/Library/Preferences/io.playcover.PlayCover.plist",
> 
>     "~/Library/Saved Application State/io.playcover.PlayCover.savedState",
> 
>     "~/Library/Application Support/me.playcover.PlayCover",
> 
>     "~/Library/Caches/me.playcover.PlayCover",
> 
>     "~/Library/Containers/me.playcover.PlayCover",
> 
>     "~/Library/Preferences/me.playcover.PlayCover.plist",
> 
>     "~/Library/Saved Application State/me.playcover.PlayCover.savedState",
> ```

ok, i will try this now,

**Comment by youthyan** [comment]
> maybe do a clean reinstall of PlayCover by deleting all the folders at these locations?

I reinstalled PlayCover by deleting all folders in these locations, and other apps also reinstalled, but it still quit unexpectedly，Except for the Genshin Impact Can be used normally。

**Comment by youthyan** [comment]
I found. Even after uninstalling Genshin Impact, if PlayTools.framework is not removed. Genshin Impact is also available

**Comment by youthyan** [comment]
After deleting and reinstalling all of them, only Genshin Impact Games can be used normally, and other apps quit unexpectedly.

**Comment by youthyan** [comment]
what should I do? I can provide any error message you want to see. I can also provide remote viewing of the computer. I can use teamviewer.

**Comment by youthyan** [comment]
> Your YouTube IPA appears to be corrupted, as it is trying to load a dylib that doesn't exist (Check the `Frameworks` folder inside `YouTube.app`)
> 
> It is trying to load `widevine_cdm_secured_ios` and fail

what should I do? I can provide any error message you want to see. I can also provide remote viewing of the computer. I can use teamviewer.

**Comment by youthyan** [comment]
It's solved, it's not the problem of playcover, nor the problem of m2, it's the problem of my computer environment. When I found that after I uninstalled playcover and deleted all these folder locations, Yuanshen can still run normally, and I knew that playcover could not be deleted. Clean, so I created a new user and group on the mac, and then installed playcover. Other apps installed can be used normally. So I don't know now, how can I delete clean palycover
